crates/bevy_animation/src/lib.rs

```diff
@@ -1171,3 +1322,291 @@ impl AnimationEvaluationState {
         Ok(())
     }
 }
+
+/// All the events from an [`AnimationClip`] that occurred this tick.
+#[derive(Debug, Clone)]
+struct TriggeredEvents<'a> {
+    direction: TriggeredEventsDir,
```

Comment:
Nit: This could probably have a better name than `new` considering how it's being constructed.